src/document/mod.rs

```diff
@@ -166,7 +168,16 @@ impl Document {
         let mut result = String::new();
         match code_blocks.get(&entrypoint) {
             Some(blocks) => {
+                let mut block_count: HashMap<&Option<String>, usize> = HashMap::new();
                 for (idx, block) in blocks.iter().enumerate() {
+                    let index = match block_count.entry(&block.name) {
```

Comment:
I think something like
```
let count = block_count.entry(&block.name).or_default();
*count += 1;
count - 1
```
might be a bit easier to read.

Reply:
(In any case, the `Occupied` branch should read `*entry.get_mut() += 1;`)

Reply:
With the last version, I get no return value (which is the old value, before `insert`). But I will try to understand the first verion, first.

Reply:
Both `or_default()` and `get_mut()` yield a `&mut usize` which can be used to both modify the value in-place and query its current value by just dereferencing. The variant the "returns" `*count - 1` is just avoid doing something like
```
let old_count = *count;
*count += 1;
old_count
```
because the compiler will transform `count - 1` into something like this anyway.

Reply:
Ok, cool! also changed in another place where it was expressed even more complicated.